src/main/scala/is/hail/utils/ArrayBuilder.scala

```diff
@@ -36,6 +36,10 @@ class ArrayBuilder[@specialized T](initialCapacity: Int)(implicit tct: ClassTag[
     size_ += 1
   }
 
+  def ++=(x: Traversable[T]) {
+    x.foreach(t => this.+=(t))
```

Comment:
I think this shouldn't be allowed.  T will get boxed in Traverable, and this interface is intended to be box-free.